modules/appeals_api/app/controllers/appeals_api/application_controller.rb

```diff
@@ -35,6 +36,12 @@ def sunset_date
 
     DEFAULT_HEADERS = { 'Content-Language' => 'en-US' }.freeze
 
+    def require_gateway_source
+      # In deployed environments, the `source` parameter is filled in by the gateway - we only
+      # want to accept requests that come through the gateway:
+      raise Common::Exceptions::Unauthorized if Rails.env.production? && params[:source].blank?
+    end
```

Comment:
This code looks good at first glance, but have you confirmed that we definitely receive a `source` parameter? I checked our controller logs in Datadog and didn't see it listed alongside our params like `id`, `icn`, `receiptDate`, etc., which concerned me. Is it there but just being excluded from our logs?

Reply:
@caseywilliams Let's put this behind a feature flag so that we can disable it should things go wonky. Default to false and we can turn it on in different envs to ensure Prod isn't affected until we've tested it.